bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/TriggerGCService.java

```diff
@@ -71,7 +71,28 @@ public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
                 Map<String, Object> configMap = JsonUtil.fromJson(requestBody, HashMap.class);
                 Boolean forceMajor = (Boolean) configMap.getOrDefault("forceMajor", null);
                 Boolean forceMinor = (Boolean) configMap.getOrDefault("forceMinor", null);
-                bookieServer.getBookie().getLedgerStorage().forceGC(forceMajor, forceMinor);
+                Double majorCompactionThreshold = (Double) configMap.getOrDefault("majorCompactionThreshold",
+                        conf.getMajorCompactionThreshold());
+                Double minorCompactionThreshold = (Double) configMap.getOrDefault("minorCompactionThreshold",
+                        conf.getMinorCompactionThreshold());
+                Long majorCompactionMaxTimeMillis =  configMap.get(
+                        "majorCompactionMaxTimeMillis") == null ? conf.getMajorCompactionMaxTimeMillis()
+                        : Long.valueOf((Integer) configMap.get("majorCompactionMaxTimeMillis"));
+                Long minorCompactionMaxTimeMillis =  configMap.get(
```

Comment:
We also need to make sure `minorCompactionMaxTimeMillis` and `majorCompactionMaxTimeMillis` are greater than `gcWaitTime` and add a unit test for this case.

Reply:
Why ? it's  not a scheduled task.